h2o-persist-s3/src/main/java/water/persist/PersistS3.java

```diff
@@ -65,11 +69,47 @@ public static AmazonS3 getClient() {
    */
   public static class H2OAWSCredentialsProviderChain extends AWSCredentialsProviderChain {
     public H2OAWSCredentialsProviderChain() {
-      super(new H2OArgCredentialsProvider(),
-          new InstanceProfileCredentialsProvider(),
-          new EnvironmentVariableCredentialsProvider(),
-          new SystemPropertiesCredentialsProvider(),
-          new ProfileCredentialsProvider());
+      super(constructProviderChain());
+
+    }
+
+    private static List<AWSCredentialsProvider> constructProviderChain() {
+      final List<AWSCredentialsProvider> providers = new ArrayList<>();
+
+      providers.add(new H2ODynamicCredentialsProvider());
+      providers.add(new H2OArgCredentialsProvider());
+      providers.add(new InstanceProfileCredentialsProvider());
+      providers.add(new EnvironmentVariableCredentialsProvider());
+      providers.add(new SystemPropertiesCredentialsProvider());
+      providers.add(new ProfileCredentialsProvider());
+
+      return providers;
+
+    }
+  }
+  
+
+  /**
+   * Holds basic credentials (Secret key ID + Secret access key) pair.
+   */
+  private static final class H2ODynamicCredentialsProvider implements AWSCredentialsProvider {
+
+
+    @Override
+    public AWSCredentials getCredentials() {
+      BufferedString keyId = DKV.getGet(S3_SECRET_KEY_ID_DKV_KEY);
```

Comment:
Sorry for nitpicking but it would be better to store this in a single object. To guarantee consistency.